kong/db/schema/init.lua

```diff
@@ -64,6 +64,9 @@ local validation_errors = {
   ENTITY_CHECK              = "failed entity check: %s(%s)",
   ENTITY_CHECK_N_FIELDS     = "entity check requires %d fields",
   CHECK                     = "entity check failed",
+  CONDITIONAL               = "failed conditional validation",
+  AT_LEAST_ONE_OF           = "at least one of these fields must be non-empty: %s",
+  ONLY_ONE_OF               = "only one of these fields must be non-empty: %s",
```

Comment:
I am slightly confused because I am not seeing `ONLY_ONE_OF` being used anywhere, and `CONDITIONAL` being tested nowhere I think? :fearful: Was `IF_THEN` missing then?
I am afraid I am either missing something, or that this was your intent? I don't know. Could you help clarifying this for me? I am not necessarily seeing this as a blocker, but rather confused and looking for what your intent here is with those fields, and what that means with regards to the maintainability of the schema validation. Or if this is future errors, or anything else really.

Reply:
they are used here: https://github.com/Kong/kong/pull/3364/files#diff-7135c27fc8fde710af0a85f4dc622d13R757

`IF_THEN` was indeed missing, but it mostly harmless because there's a fallback `"failed entity check: %s(%s)"` message (which adds further data). The idea is that entity checkers (added by us or in the future by others) will always produce messages, whether they return a custom message, a default message, or just a boolean.

Reply:
(and just to clarify, the `conditional` entity checker is tested)

Reply:
Oh I see, it is tested but does not have an assertion for its error message content. Alright, good enough it seems unrelated to the scope of this PR anyway.